static/js/components/UpdateSourceRedshift.jsx

```diff
@@ -33,6 +33,10 @@ const UpdateSourceRedshift = ({ source }) => {
   const [state, setState] = useState(
     source.redshift ? String(source.redshift) : ""
   );
+  const [uncstate, setUncState] = useState(
```

Comment:
Actually, probably best to combine redshift and error into a single state object. Then add relevant "name" attrs to the input fields, and update using event.target.name as the state object key (see https://github.com/skyportal/skyportal/blob/master/static/js/components/NewGroupForm.jsx#L60-L67)

Reply:
Great suggestion.  Forced me to learn more about react and update my firefox after 8 years of neglect!